Ush 1445 - Phone Input Field (Mobile)

ushahidlee opened the pull request:
**Issue:**

Our users would like a nice field specific to the entry of phone numbers.

**Solution:**

This PR wraps a widely used Javascript phone input library, with a Typescript Angular component so as to work inside of our ionic mobile app.

IMPORTANT: This PR has been implemented _BEFORE_ the associated backend ticket and cannot be tested on the system at large until the backend ticket has been committed. The testing plan below can be employed until that time.

**Testing:**

1. Open the mobile app
2. Add the deployment berlincalling.ushahidi.io
3. Create a post using the "Basic Post" survey.
(The last field in the survey is the mobile input field.)
4. Create a post with a phone number.
- [x] Only numbers are accepted.
- [x] Number length is limited to country standard
- [x] Country code can be manually (using the dropdown) and automatically selected (by typing + and then the country phone prefix)
5. Edit the post.
- [x] Number and Country are maintained.
- [x] Changes are applied.

ushahidlee committed:
changes all over the place

ushahidlee committed:
tiny changes

ushahidlee committed:
getting there

ushahidlee committed:
ch ch ch changes

ushahidlee committed:
fixed state issues

ushahidlee committed:
fix country

ushahidlee committed:
delete logging

ushahidlee committed:
fixed country code duplication

ushahidlee committed:
Ush 1299 - Upgrade to Capacitor 6 (Android 14) (#1184)

* Capacitor 6

* Fix for capacitor camera bug

ushahidlee committed:
dark mode

ushahidlee committed:
updated dependency and fixed strictness

MMwandigha committed:
Checking user permissions for collections (#1190)

MMwandigha committed:
Removing toast messages during startup (#1188)

Angamanga commented on apps/mobile-mzima-client/src/app/post/post-edit/post-edit.page.html:
```diff
@@ -69,7 +69,7 @@ <h2 class="task-label">{{ task.label }}</h2>
           <ng-container
-            *ngIf="field.input === 'text' && (field.type === 'title' || field.type === 'varchar')"
+            *ngIf="field.input === 'text' && (field.type === 'title' || field.type === 'varchar') && field.label !== 'mobile_test'"
           >
```
Is this for the temporary test-thing?

ushahidlee replied:
yeah exactly. The new field will be a 'text' field in the backend, so its a very close simulation of end to end.

Angamanga reviewed (approved):
Works fine as far as I could test it 👍

Ifycode committed:
Fix: Use date pipe instead and include source & status on postcards

Ifycode committed:
Fix | functionality: children category UI and filters fix for mobile

Ifycode committed:
Fix: UI for date, source & status on postcards

Ifycode committed:
Fix: Translation for source area on postcards

Ifycode committed:
Fix | UI: categories filters dropdown UI restructuring

Ifycode committed:
Fix: Subcategory count fix, and enable apply button on check or uncheck of subcategory

Ifycode committed:
Fix: Remove no longer used subcategories modal code

Ifycode committed:
Fix: Dropdown functionality - show and hide subcategories on parent click

Ifycode committed:
Fix: Add no sub category description & translate all parent category descriptions

Ifycode committed:
Fix: Set & add colors for category parents name avatar

Ifycode committed:
Fix: Use existing avatar color library/code instead

Ifycode committed:
Fix: Translate and fix broken placeholder and hint signup/login input texts

MMwandigha committed:
[USH-1539] Collection-editing rights disappear in development after refreshing the mobile app on web (#1364)

* Fix for permission rights disappearing after refresh

* Fixed collections for non admin users with permissions

MMwandigha committed:
[USH-1402] Collections created with Only Me visibility cannot have the visibility changed (#1362)

* Allowing collection visibility to change after only-me

* Returning back commented line

ushahidlee committed:
Fixed radio option wrapping bug (#1372)

ushahidlee committed:
updated version for release

MMwandigha committed:
[USH-1443] First page of posts not seen on navigating back to My posts page from viewing a post (#1380)

* pagination attempted fix for my posts

* Added comments for context

Angamanga committed:
Unhiding language-switcher and adding more languages (#1388)

* Unhiding language-switcher in mobile app

* Adding more language-strings

* Adding Swahili

* Removing logs

ushahidlee committed:
sync package json and lock file

ushahidlee committed:
Ush 1637 - Allow backend_url be specified in mobile app when not in production (#1419)

* hack to simplify development

transifex-integration[bot] committed:
Translate en.json in cs (#1457)

81% of minimum 80% translated source file: 'en.json'
on 'cs'.

Sync of partially translated files: 
untranslated content is included with an empty translation 
or source language content depending on file format

Co-authored-by: transifex-integration[bot] <43880903+transifex-integration[bot]@users.noreply.github.com>

transifex-integration[bot] committed:
Translate en.json in sw (#1458)

81% of minimum 80% translated source file: 'en.json'
on 'sw'.

Sync of partially translated files: 
untranslated content is included with an empty translation 
or source language content depending on file format

Co-authored-by: transifex-integration[bot] <43880903+transifex-integration[bot]@users.noreply.github.com>
Co-authored-by: Anna Iosif <anna.iosif@me.com>

ushahidlee committed:
Ush 1589 - View Multimedia Content (Mobile) (#1422)

MMwandigha committed:
Allow all collections to be edited and not only the user's collections (#1459)

MMwandigha committed:
Changing the text for 'related post' survey field on mobile (#1470)

MMwandigha committed:
Show 'fetching surveys' instead of 'getting data' for loading surveys (#1466)

MMwandigha committed:
Mobile: Changing wording in success message after creating/editing post (#1473)

Ifycode committed:
[USH-1511] Use only parameters for posts in Mobile (#1455)

* Fix: Only parameters for posts - also fixes preview for image & twitter

* Fix: import & use shared helper variable (that was added in USH-1510 web ticket)

---------

Co-authored-by: Anna Iosif <anna.iosif@me.com>

Ifycode committed:
Fix: Add only parameters for collections in mobile

Ifycode committed:
Fix: Add only parameters for categories in mobile

tuxpiper committed:
Update package-lock.json

Angamanga committed:
Using only-parameters for surveys + other performance issue (#1504)

* Using ONLY-parameters for surveys

* Remove console.log

Angamanga committed:
Hiding Intercom functionality in mobile app (#1511)

* Removing intercom-window from app

* Adjustment to text

tuxpiper committed:
mobile-v2.202503.0

MMwandigha committed:
Adjusting map view fab position for iOS devices (#1512)

MMwandigha committed:
fix for missing localization keys & inner html for 'no posts found' (#1517)

MMwandigha committed:
setting media fields value as an array, enhancing photo validation to support arrays (#1523)

tuxpiper commented:
I’ve pulled it up to be based on the latest of mobile-development . 

The only conflicts were with packages.json  and -lock.json . Those files I overwrote with the latest of mobile-development, so they would need to be reviewed again here if changes need to be re-introduced.

tuxpiper commented:
Changes on package.json after the rebase:
```
diff --git a/package.json b/package.json
index afdb56558..f79238314 100644
--- a/package.json
+++ b/package.json
@@ -30,25 +30,26 @@
     "@angular/compiler": "~14.2.7",
     "@angular/core": "~14.2.7",
     "@angular/forms": "~14.2.7",
+    "@angular/material": "^14.2.7",
     "@angular/platform-browser": "~14.2.7",
     "@angular/platform-browser-dynamic": "~14.2.7",
     "@angular/router": "~14.2.7",
     "@asymmetrik/ngx-leaflet": "^14.0.1",
     "@asymmetrik/ngx-leaflet-markercluster": "^14.0.1",
-    "@capacitor-community/intercom": "^5.0.1",
-    "@capacitor/android": "^5.7.4",
-    "@capacitor/app": "^5.0.7",
-    "@capacitor/camera": "^5.0.9",
-    "@capacitor/cli": "^5.7.4",
-    "@capacitor/core": "^5.7.4",
-    "@capacitor/device": "^5.0.7",
-    "@capacitor/dialog": "^5.0.7",
-    "@capacitor/filesystem": "^5.2.1",
-    "@capacitor/geolocation": "^5.0.7",
-    "@capacitor/keyboard": "^5.0.8",
-    "@capacitor/network": "^5.0.7",
-    "@capacitor/share": "^5.0.7",
-    "@capacitor/status-bar": "^5.0.7",
+    "@capacitor-community/intercom": "^6.0.0",
+    "@capacitor/android": "^6.1.2",
+    "@capacitor/app": "^6.0.0",
+    "@capacitor/camera": "^6.0.2",
+    "@capacitor/cli": "^6.1.2",
+    "@capacitor/core": "^6.1.2",
+    "@capacitor/device": "^6.0.0",
+    "@capacitor/dialog": "^6.0.0",
+    "@capacitor/filesystem": "^6.0.0",
+    "@capacitor/geolocation": "^6.0.0",
+    "@capacitor/keyboard": "^6.0.1",
+    "@capacitor/network": "^6.0.1",
+    "@capacitor/share": "^6.0.1",
+    "@capacitor/status-bar": "^6.0.0",
     "@ionic/angular": "^7.1.1",
     "@ionic/pwa-elements": "^3.1.1",
     "@ionic/storage-angular": "^4.0.0",
@@ -60,16 +61,15 @@
     "@supy-io/ngx-intercom": "^14.2.12",
     "@swimlane/ngx-charts": "^20.1.2",
     "angular-google-tag-manager": "^1.6.1",
-    "capacitor-native-settings": "^5.0.1",
+    "capacitor-native-settings": "^6.0.1",
     "dayjs": "^1.11.7",
-    "intl-tel-input": "^23.8.1",
     "ion2-calendar": "^3.5.0",
     "ionicons": "^7.1.0",
     "leaflet": "^1.9.3",
     "leaflet-control-geocoder": "^2.4.0",
     "leaflet.markercluster": "^1.5.3",
     "leaflet.offline": "^3.0.0",
-    "libphonenumber-js": "^1.11.5",
+    "locale-codes": "^1.3.1",
     "lodash": "^4.17.21",
     "lottie-web": "^5.10.2",
     "masonry-layout": "^4.2.2",
@@ -78,7 +78,6 @@
     "ngx-custom-tour": "^1.1.2",
     "ngx-lottie": "^9.1.0",
     "ngx-masonry": "^14.0.1",
-    "ngx-mat-intl-tel-input": "^5.0.0",
     "ngx-pagination": "^6.0.3",
     "ngx-quill": "^19.0.1",
     "ngx-sharebuttons": "^11.0.0",
```
